developer-docs-site/docs/tutorials/move-tutorial/README.md

```diff
@@ -0,0 +1,689 @@
+  │ 0xc0ffee:
+  │       => key 0xcafe::basic_coin::Coin {
+  │           value: 10
+  │       }
+  └──────────────────
+  ```
+* Find a flag that allows you to gather test coverage information, and
+  then play around with using the `move coverage` command to look at
+  coverage statistics and source coverage.
+
+</details>
+
+## Step 3: Designing my `basic_coin` module<span id="Step3"><span>
+
+In this section, we are going to design a module implementing a basic coin and balance interface, where coins can
+be minted and transferred between balances held under different addresses.
+
+The signatures of the public Move function are the following:
+
+```
+/// Publish an empty balance resource under `account`'s address. This function must be called before
+/// minting or transferring to the account.
+public fun publish_balance(account: &signer) { ... }
+
+/// Mint `amount` tokens to `mint_addr`. Mint must be approved by the module owner.
+public fun mint(module_owner: &signer, mint_addr: address, amount: u64) acquires Balance { ... }
+
+/// Returns the balance of `owner`.
+public fun balance_of(owner: address): u64 acquires Balance { ... }
+
+/// Transfers `amount` of tokens from `from` to `to`.
+public fun transfer(from: &signer, to: address, amount: u64) acquires Balance { ... }
+```
+
+Next we look at the data structs we need for this module.
+
+A Move module doesn't have its own storage. Instead, Move "global storage" (what we call our
+blockchain state) is indexed by addresses. Under each address there are Move modules (code) and Move resources (values).
+
+The global storage looks roughly like this in Rust syntax:
+
+```rust
+struct GlobalStorage {
+    resources: Map<address, Map<ResourceType, ResourceValue>>
+    modules: Map<address, Map<ModuleName, ModuleBytecode>>
+}
+```
+
+The Move resource storage under each address is a map from types to values. (An observant reader might observe that
+this means each address can only have one value of each type.) This conveniently provides us a native mapping indexed
+by addresses, which is by constructionn type-safe. In our `basic_coin` module, we define the following `Balance` resource representing the number of coins
+each address holds:
+
+```
+/// Struct representing the balance of each address.
+struct Balance has key {
+    coin: Coin // same Coin from Step 1
+}
+```
+
+Roughly the Move blockchain state should look like this:
+
+![](diagrams/move_state.png)
+
+#### Advanced topics:
+<details>
+<summary><code>public(script)</code> functions</summary>
+
+Only functions with the `entry` qualifier can be invoked directly in transactions. So if you would like to call the `transfer`
+method directly from a transaction, you'll want to change its signature to:
+```
+public entry fun transfer(from: signer, to: address, amount: u64) acquires Balance { ... }
+```
+Read more on Move function visibilities [here](https://move-language.github.io/move/functions.html#visibility).
```

Comment:
@clay-aptos we do have a lot of links from the tutorial into the Move book. What would be canonical URL for the book in our docset?

Reply:
> @clay-aptos we do have a lot of links from the tutorial into the Move book. What would be canonical URL for the book in our docset?

Hi @wrwg , I have fixed all of the links to be relative and go to our version of the Move Book in direct commits here. Please note, the links to the `coin` subdirectories are broken as they have been renamed from `basic_coin` to `BasicCoin` it seems.

Regardless, we should fully qualify them to ensure they work from Aptos.dev, meaning linking to the live version of:
https://github.com/aptos-labs/aptos-core/tree/dae90b203ed10d2110b58e453a52533736b96fe3/developer-docs-site/docs/tutorials/move-tutorial/step_1/BasicCoin

Or as I suggested in the issue comment below:
https://github.com/aptos-labs/aptos-core/issues/6375#issuecomment-1451145070

We may want to move them out of the `developer-docs-site` directory and perhaps locate them in:
https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/move-examples/

I can take care of fixing those links while editing the PR once we decide where those files will exist. Let me know what you think. Thanks!

Reply:
Thanks for fixing the links @clay-aptos!

I think it is actually a great idea to move this into `move-examples`. It is designed for github _or_ you can actually have a local copy of the .md (in a github client) and just browse it directly in an IDE. That's why the paths should stay relative. This is more like a "codelab" than a "tutorial" where people are expected to actively play with the sources.

Reply:
> Thanks for fixing the links @clay-aptos!
> 
> I think it is actually a great idea to move this into `move-examples`. It is designed for github _or_ you can actually have a local copy of the .md (in a github client) and just browse it directly in an IDE. That's why the paths should stay relative. This is more like a "codelab" than a "tutorial" where people are expected to actively play with the sources.

Wonderful, @wrwg !  The only issue is that I don't think relative links will work outside of the `developer-docs-site` directory once the site is built for Aptos.dev since the files won't be included. (They will work at GitHub.com and in IDEs.) This is why we recommend using fully qualified URLs for files outside of the site directory:
https://aptos.dev/community/aptos-style#form-links-properly

Please let me know if you want me to move the `step` directories to a new directory in `move-examples`. I can try to do that as a commit here or create another PR to do so, assuming you are alright with linking to them via GitHub.com URLs.

We could also try to include relevant sections of the source files directly as we are using on the tutorials David mentioned. Let me know what you think. Thanks!

Reply:
I’m dealing with the move, I let you know once I need more helo